File: template/app/src/main/java/com/xmartlabs/template/model/common/BuildInfo.java
package com.xmartlabs.template.model.common;

import com.annimon.stream.Objects;
import com.xmartlabs.template.BuildConfig;

public class BuildInfo implements com.xmartlabs.bigbang.core.model.BuildInfo {
  @Override
  public boolean isDebug() {
    return BuildConfig.DEBUG;
  }

  @Override
  public boolean isStaging() {
    return Objects.equals(BuildConfig.FLAVOR, BuildType.STAGING);
  }

  @Override
  public boolean isProduction() {
    return Objects.equals(BuildConfig.FLAVOR, BuildType.PRODUCTION);
  }
}

File: template/app/src/main/java/com/xmartlabs/template/model/common/BuildInfo.kt
package com.xmartlabs.template.model.common

import com.xmartlabs.template.BuildConfig
import com.xmartlabs.bigbang.core.model.BuildInfo as CoreBuildInfo

class BuildInfo : CoreBuildInfo {
  override val isDebug = BuildConfig.DEBUG
  override val isProduction: Boolean
    get() = BuildConfig.FLAVOR == BuildType.STAGING.toString()
  override val isStaging: Boolean
    get() = BuildConfig.FLAVOR == BuildType.PRODUCTION.toString()
}
